app/controllers/lesson_plans_controller.rb

```diff
@@ -56,6 +56,34 @@ def update
     end
   end
 
+  def create_lesson
+    params[:lesson_plan] = {
+      lesson_plan_lessons_attributes: [
+        { lesson_id: params[:lesson_id] }
+      ]
+    }
+
+    if helpers.current_lesson_plan.persisted?
+      update
+    else
+      create
+    end
+  end
+
+  def destroy_lesson
+    lesson = Lesson.find(params[:lesson_id])
+    lesson_plan_lesson = helpers.current_lesson_plan_lesson(lesson)
```

Comment:
Calling this helper in the view seems a bit odd to be. It seems like we are having to do some tricks to coerce these submissions into the right format for a call to `update`.

I wonder if we should either build the full form in the view or let this be a standalone controller that does lesson_plan.lessons.delete(params[:lesson_id) or something similar.

FWIW, I'd be find with not supporting javascript for lesson plans. I think that users without js would still be able to access the core functionality of the site.

In any case, I think we could mandate on the model that lessons be unique within lesson plans, since the UI expects that.

Reply:
:+1: these actions could be simpler.